sway/commands/bind.c

```diff
@@ -209,3 +268,20 @@ struct cmd_results *cmd_bindsym(int argc, char **argv) {
 struct cmd_results *cmd_bindcode(int argc, char **argv) {
 	return cmd_bindsym_or_bindcode(argc, argv, true);
 }
+
+
+/**
+ * Execute the command associated to a binding
+ */
+void seat_execute_command(struct sway_seat *seat, struct sway_binding *binding) {
+	wlr_log(WLR_DEBUG, "running command for binding: %s",
+		binding->command);
+	config->handler_context.seat = seat;
+	struct cmd_results *results = execute_command(binding->command, NULL);
+	transaction_commit_dirty();
```

Comment:
This call to `transaction_commit_dirty` shouldn't be here. It's already called in `dispatch_cursor_button`.

Reply:
I've removed it from `seat_execute_command`, and placed it at the end of `handle_keyboard_key`, as a cleanup point, following the example of `dispatch_cursor_button`. (Tracking precisely which branches can cause changes ends up brittle.)